### ISSUE TITLE
Game Freezes due to AI Scripting issues

### ISSUE THREAD

**Issue by Zombie-Ryushu** [Wargus/war1gus#89]
It seems Stratagus freezes when it tries to issue orders to computer controlled units it logically can't. Stratagus behaves this way often./

**Comment by Zombie-Ryushu** [comment]
This has been a peristent problem, where something with the AI Scripting causes Stratagus to engage in behavior that uses 100% of a core and hangs the game.

**Comment by timfel** [comment]
Yes, I havent spent enough time with the ai. The issue is usually that the ai tries to build something (like a farm) and cannot find a building place (because not enough roads are available or something like that). Unfortunately these conditions are hard to check from the lua scripts.

**Comment by Zombie-Ryushu** [comment]
I need some larger maps (128x128) to stress the A/I and to observe the behavior of Cross Race forces. Do you have some?

**Comment by timfel** [comment]
I don't, unfortunately

**Comment by Zombie-Ryushu** [comment]
Can you compose one? I've noticed some strange behaviours with forces containing Human Peasants in Orc Forces. I'd also like to see if we can't do  an update to allow Campaign forces to be more agressive.

**Comment by timfel** [comment]
Oh, yes, the AI cannot deal with such mixes - the AI scripts check what race the AI should have, and then only considers upgrades, buildings, and units from that race. If you give a peasant to an orc AI, I don't know what would happen.

**Comment by Zombie-Ryushu** [comment]
He made farms but nothing else, but thats not the only situation I have seen, I've also seen the AI "surrender" cease and desist agressive actions. Reverting to only gathering resources.

**Comment by timfel** [comment]
Yes, that makes sense. Farms are not explicitly requested in the Ai script, but rather the AI just goes and checks if it has any units that can build buildings that increase supply - for that, it would use the peasant, but for nothing else. Also, the AI would not use any other opposite-race units for attacking, if you give 100 footmen to an orc AI, they will just stand there and do nothing unless attacked.

**Comment by Zombie-Ryushu** [comment]
ne of the bugs is that Farms do not correctly display Resource consumption. They always say 5 and 0. There have been a few instances where I have made Warcraft II Maps with dual race forces by making two forces the same color.

**Comment by Zombie-Ryushu** [comment]
Would it be possible to make Cross Race scripts that would not damage single race scripts?

**Comment by timfel** [comment]
Possible, just quite some work ;)

**Comment by Zombie-Ryushu** [comment]
What about a Switch that would allow Campaign levels to act like Solo levels and have the computer act more aggressive and expand?
